logging/impl.go

```diff
@@ -106,7 +106,11 @@ func (imp *impl) AsZap() *zap.SugaredLogger {
 	config := NewZapLoggerConfig()
 	// Use the global zap `AtomicLevel` such that the constructed zap logger can observe changes to
 	// the debug flag.
-	config.Level = GlobalLogLevel
+	if imp.level.Get() == DEBUG {
```

Comment:
is this an issue specifically with DEBUG? or do we need to add guards for WARN/ERROR as well?